### PR TITLE
[VUFIND-1593] Add an HTML-friendly JSON encode helper.

### PR DESCRIPTION
**Pull request by demiankatz**
TODO:
- [x] Test everything
- [x] Close [VUFIND-1593](https://openlibraryfoundation.atlassian.net/browse/VUFIND-1593) when merging

### CODE REVIEW

**Comment by demiankatz** [comment]
@EreMaijala, I have a bit more testing to do, but I'm interested in your thoughts on this approach. A few notes:

1.) I came up with a better fix for the channel escaping than #2687 -- I figured out where the double-escaping was coming from (the URL query helper in the channel provider plugin).

2.) I assume that I don't need to escape JSON differently inside inline scripts, so I just focused on JSON in HTML attributes.

3.) For consistency, a view helper seemed like the best approach... but I'm open to other ideas!

**Comment by demiankatz** [comment]
At this point, all integration tests are passing, though there are some less-used templates (like WorldCatIdentities) that probably need to be manually double-checked. I'll wait to finish all the manual testing until the general approach has been approved, so I only have to do it once. :-)

File: module/VuFind/src/VuFind/View/Helper/Root/HtmlSafeJsonEncode.php
Comment (EreMaijala) [inline module/VuFind/src/VuFind/View/Helper/Root/HtmlSafeJsonEncode.php]: Could you use the `($this->getView()->plugin($outerEscaper))($json)` style call here?

File: module/VuFind/src/VuFind/View/Helper/Root/HtmlSafeJsonEncode.php
Comment (EreMaijala) [inline module/VuFind/src/VuFind/View/Helper/Root/HtmlSafeJsonEncode.php:4]: This could include a short description on why this is needed.

**Review by EreMaijala — CHANGES_REQUESTED** [review]
Makes sense to me! Just a couple of tiny nitpicks in comments.

**Comment by demiankatz** [comment]
Thanks, @EreMaijala! I've incorporated your feedback, manually tested things, and added a unit test for the new helper. Hopefully this is ready to go now!

**Review by EreMaijala — APPROVED** [review]

**Pull request merged**
